### PR TITLE
Having a text file of binary-matrix

### PR DESCRIPTION
**Pull request by soudabeh19**
Adding text_files function to have :
row_index.txt
column_index.txt
matrix.txt

### CODE REVIEW

File: verifyFiles.py [hunk 1]
```diff
@@ -366,22 +367,43 @@ def Ldiff_print(Diff,conditions_dict):
...
-  #       flag=True
-  #       while flag:
-  #           j=len (path_list[0][:])-1 # find the number of elements in path_list to be printed (except the st_mtime)
-  #           print path_list[i][0:j]
-  #           flag=False
-  #  	     i+=1
-    #------------ Print the Matrix -----------------------
-    print " >>> Conditions order : ",bDiff.keys() 
+            path_list.append([subject,path,Ldiff[subject][path],first_subject[path].st_mtime])   
+    print "\n\n  >>> Conditions order : ",bDiff.keys(), "\n" # letting know about order of values in binary matrix dataframe
```
Comment (glatard) [inline verifyFiles.py]: keys in a dictionary are not ordered.

File: verifyFiles.py [hunk 2]
```diff
@@ -340,6 +340,7 @@ def Ldiff_print(Diff,conditions_dict):
     bDiff=Diff
     No_pair_con=len(bDiff.keys())
     Ldiff={}
+    tSpark={}
```
Comment (glatard) [inline verifyFiles.py]: Remove if unused.

File: verifyFiles.py [hunk 3]
```diff
@@ -366,22 +367,43 @@ def Ldiff_print(Diff,conditions_dict):
...
+            path_list.append([subject,path,Ldiff[subject][path],first_subject[path].st_mtime])   
+    print "\n\n  >>> Conditions order : ",bDiff.keys(), "\n" # letting know about order of values in binary matrix dataframe
     df = pd.DataFrame([[col1,col2,col3] for col1, d in Ldiff.items() for col2, col3 in d.items()],columns=['Subject','File','Results'])
-    pd.set_option('display.max_rows', None)    
+    pd.set_option('display.max_rows', None) # display the dataframe of binary matrix
+    text_files (bDiff,conditions_dict)
     return df
 
+# making output textfile of the binary matrix (matrix.txt, row_index.txt, column_index.txt)  
+def text_files (bDiff,conditions_dict):
```
Comment (glatard) [inline verifyFiles.py]: rename to "write_text_files"

File: verifyFiles.py [hunk 4]
```diff
@@ -366,22 +367,43 @@ def Ldiff_print(Diff,conditions_dict):
...
-  #           flag=False
-  #  	     i+=1
-    #------------ Print the Matrix -----------------------
-    print " >>> Conditions order : ",bDiff.keys() 
+            path_list.append([subject,path,Ldiff[subject][path],first_subject[path].st_mtime])   
+    print "\n\n  >>> Conditions order : ",bDiff.keys(), "\n" # letting know about order of values in binary matrix dataframe
     df = pd.DataFrame([[col1,col2,col3] for col1, d in Ldiff.items() for col2, col3 in d.items()],columns=['Subject','File','Results'])
-    pd.set_option('display.max_rows', None)    
+    pd.set_option('display.max_rows', None) # display the dataframe of binary matrix
+    text_files (bDiff,conditions_dict)
```
Comment (glatard) [inline verifyFiles.py]: Put this under the `-d` option, see https://github.com/big-data-lab-team/repro-tools/issues/52

**Review by glatard — CHANGES_REQUESTED** [review]
See comments inline as discussed during meeting.

**Comment by lalet** [comment]
@soudabeh19 Tests are failing since there are changes in the parameters. Tests need to be edited.

File: verifyFiles.py [hunk 5]
```diff
@@ -537,17 +553,12 @@ def main():
 	#and the file checksumFile,checkCorruption and the path to the sqlite file.
         #diff,metric_values,dictionary_executables,dictionary_processes=n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_from_file_dict,args.checksumFile,args.checkCorruption,args.sqLiteFile)i
 	diff,bDiff,metric_values,dictionary_executables,dictionary_processes,metric_values_subject_wise=n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_from_file_dict,args.checksumFile,args.checkCorruption,args.sqLiteFile,args.trackProcesses)
-       	if args.fileDiff is not None:
-            log_info("Writing difference matrix to file "+args.fileDiff)
-            diff_file = open(args.fileDiff,'w')
+       	if args.difference is not None:
+            log_info("Writes the difference matrices and indexs into files"+args.difference)
```
Comment (lalet) [inline verifyFiles.py:557]: args.difference is the base name of the file. It would be better to edit the log info to a  more meaningful one.

**Review by lalet — APPROVED** [review]
Good work @soudabeh19 ! Approved. Minor changes are needed.

File: verifyFiles.py [hunk 6]
```diff
@@ -495,13 +512,12 @@ def main():
                                              /home/$(USER)/CentOS7.FSL5.0.6
                                              Each directory will contain subject folders like 100307,100308 etc'''))
         parser.add_argument("-c", "--checksumFile",help="Reads checksum from files. Doesn't compute checksums locally")
-	parser.add_argument("-d", "--fileDiff", help="Writes the difference matrix into a file")
+	parser.add_argument("difference", help="Writes the difference matrices and indexs into files")
```
Comment (glatard) [inline verifyFiles.py:515]: Help text should be "base name to use in output file names".

**Pull request merged**